hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -589,7 +589,7 @@ private void handleFlushInternal(boolean close)
       executePutBlock(true, true);
     }
     waitOnFlushFutures();
-    watchForCommit(false);
+//    watchForCommit(false);
```

Comment:
watchForCommit is used by waitForFlushAndCommit too.

RatisBlockOutputStream.hsync() calls waitForFlushAndCommit() call, which is redundant if handleFlush() is already called. Perhaps we can update it like:

      if (getBufferPool() != null && getBufferPool().getSize() > 0) {
        handleFlush(false);
      } else {
        waitForFlushAndCommit(false);
      }

Reply:
You are right. It looks redundant. Done